src/vector/v.maxent.swd/v.maxent.swd.py

```diff
@@ -257,156 +289,213 @@ def main():
+            raster_format = "AAIGrid"
+            raster_extension = "asc"
+        elif options["format"] == "GeoTIFF":
+            raster_format = "GTiff"
+            raster_extension = "tif"
+        else:
+            gs.message(
+                "Only ascii and geotif are supported. Exporting rasters as ascii"
+            )
+            raster_format = "AAIGrid"
+            raster_extension = "asc"
+        if len(evp) > 0:
+            for idx, name in enumerate(evp):
+                exporturl = os.path.join(
+                    options["export_rasters"], f"{evpn[idx]}.{raster_extension}"
+                )
+                gs.run_command(
+                    "r.out.gdal",
+                    input=name,
+                    output=exporturl,
+                    format=raster_format,
+                    nodata=int(options["nodata"]),
+                    flags="c",
+                    quiet=True,
+                )
+        if len(evpc) > 0:
+            for idx, name in enumerate(evpc):
+                exporturl = os.path.join(
+                    options["export_rasters"], f"{evpcn[idx]}.{raster_extension}"
+                )
+                gs.run_command(
+                    "r.out.gdal",
+                    input=name,
+                    output=exporturl,
+                    format=raster_format,
+                    flags="c",
+                    type="Int16",
+                    nodata=-9999,
+                    quiet=True,
+                )
 
     # --------------------------------------------------------------------------
     # Background points
     # --------------------------------------------------------------------------
 
     # Create / copy to tmp layer
-    bgpname = tmpname("bgp")
-    if bgp == "":
-        grass.run_command(
-            "r.random", input=evp[0], npoints=bgpn, vector=bgpname, quiet=True
-        )
-        grass.run_command("v.db.droptable", flags="f", map=bgpname, quiet=True)
-        grass.run_command("v.db.addtable", map=bgpname, table=bgpname, quiet=True)
+    bgpname = create_temporary_name("bgp")
+    gs.message(_("Creating SWD file with background points"))
+    if bool(options["bgp"]):
+        gs.run_command("g.copy", vector=[options["bgp"], bgpname], quiet=True)
     else:
-        grass.run_command("g.copy", vector=[bgpn, bgpname], quiet=True)
-    grass.run_command("v.db.addcolumn", map=bgpname, columns=evp_cols, quiet=True)
+        if bool(options["seed"]):
+            gs.run_command(
+                "r.random",
+                input=environmental_layers[0],
+                npoints=bgpn,
+                vector=bgpname,
+                quiet=True,
+                seed=int(options["seed"]),
+            )
+        else:
+            gs.run_command(
+                "r.random",
+                input=environmental_layers[0],
+                npoints=bgpn,
+                vector=bgpname,
+                quiet=True,
+                flags="s",
+            )
```

Comment:
You could simplify / shorten this:
```suggestion
        gs.run_command(
            "r.random",
            input=environmental_layers[0],
            npoints=bgpn,
            vector=bgpname,
            quiet=True,
            seed=int(options["seed"]) if options["seed"] else None,
        )
```

It may be enough (I guess) with:
```
seed=options["seed"],
```
But I have not tested...

Reply:
You're right, there is a default seed , so there is never a need to set the -s flag